source/common/formatter/substitution_formatter.cc

```diff
@@ -752,7 +752,8 @@ StreamInfoFormatter::StreamInfoFormatter(const std::string& field_name) {
     field_extractor_ = std::make_unique<StreamInfoStringFieldExtractor>(
         [](const StreamInfo::StreamInfo& stream_info) {
           std::string upstream_cluster_name;
-          if (stream_info.upstreamClusterInfo().has_value()) {
+          if (stream_info.upstreamClusterInfo().has_value() &&
+              stream_info.upstreamClusterInfo().value() != nullptr) {
```

Comment:
It seems unfortunate that we can have upstreamClusterInfo be set to nullptr, but this behavior is documented in upstreamClusterInfo's interface definition.

Reply:
That's a very unintuitive interface: an optional pointer that can be null. Can this be cleaned up to be not-optional, or not-nullable?